src/System.Runtime/tests/System/Guid.cs

```diff
@@ -176,4 +177,31 @@ public static void TestToString()
         Assert.Equal(_testGuid.ToString("P"), "(a8a110d5-fc49-43c5-bf46-802db8f843ff)");
         Assert.Equal(_testGuid.ToString("X"), "{0xa8a110d5,0xfc49,0x43c5,{0xbf,0x46,0x80,0x2d,0xb8,0xf8,0x43,0xff}}");
     }
+
+    [Fact]
+    public static void TestRandomness()
+    {
+        const int Iterations = 100;
+        const int GuidSize = 16;
+        byte[] random = new byte[GuidSize * Iterations];
+
+        for (int i = 0; i < Iterations; i++)
+        {
+            // Get a new Guid
+            Guid g = Guid.NewGuid();
+            byte[] bytes = g.ToByteArray();
+
+            // Make sure it's different from all of the previously created ones
+            for (int j = 0; j < i; j++)
+            {
+                Assert.False(bytes.SequenceEqual(new ArraySegment<byte>(random, j * GuidSize, GuidSize)));
+            }
+
+            // Copy it to our randomness array
+            Array.Copy(bytes, 0, random, i * GuidSize, GuidSize);
+        }
+
+        // Verify the randomness of the data in the array
+        RandomDataGenerator.VerifyRandomDistribution(random);
```

Comment:
Thanks.  I'd considered that, but the test already allows for a (relatively) huge amount of variability (7%).  I'm going to leave it as is until we find there's a problem, and then we can add some complication to compensate, probably at the same time that we improve the randomness test itself :smile: